pkg/api/vminfo.go

```diff
@@ -179,17 +179,24 @@ func (vm *VmInfo) applyBounds(b scalingBounds) {
 	vm.Mem.Max = uint16(b.Max.Mem.Value() / vm.Mem.SlotSize.Value())
 }
 
-type scalingBounds struct {
-	Min *resourceBound `json:"min,omitempty"`
-	Max *resourceBound `json:"max,omitempty"`
+// ScalingBounds is the type that we deserialize from the "autoscaling.neon.tech/bounds" annotation
+//
+// All fields (and sub-fields) are pointers so that our handling can distinguish between "field not
+// set" and "field equal to zero". Please note that all field are still required to be set and
+// non-zero, though.
```

Comment:
I feel like the error "<x> must be set to a non-zero value" is clear and actionable and it seems like the set/unset level means that a user might not set the value, set it to zero, retry and then get the next error (possibly multiplied (ish) for both fields,) when we can simplify the validation logic significantly and provide more actionable feedback to users more quickly.

Reply:
That's a really good point; I'll change that before merging this PR.

Reply:
also if they're not pointers the logic gets simpler and you can do the `ec := &erc.Collector{}; erc.When(ec, cond, "message")` to get all the errors at once to avoid the extra retry cycle

Reply:
Opened #190 to do this